target/linux/realtek/dts-5.15/rtl9302_zyxel_xgs1250-12.dts

```diff
@@ -223,55 +223,87 @@
 			phy-handle = <&phy0>;
 			phy-mode = "xgmii";
 			led-set = <0>;
+			led-num = <(
+				RTL93XX_LED_NUM0 | /* Amber */
+				RTL93XX_LED_NUM1   /* Lime */
+			)>;
```

Comment:
> `led-set` and `led-num` aren't upstreamable. This is also very RTL93xx-specific, and doesn't allow a user to reference an LED (for standard OpenWrt software control), which is needed on some devices (e.g. Cisco SG220). But good shouldn't the evil of perfect, and I'm willing to accept this as a stop-gap solution to improve device description.

Well we already had led-set, there's nothing new here, I only added led-num. And yes, they would have to be 'realtek,rtl93xx-led-set' at the least :p but I also do not intend to upstream either of those (hopefully). These are temporary, until we can go to a better driver. And then, it'll need to cover all our cases obviously.

> 
> My proposed led controller driver handled this by requiring every used LED to be defined individually. The driver would then determine how to configure the hardware in the most appropriate way. So in a sense what you're adding here is similar.

Yeah, I saw that, but then, I don't like that either. While this is certainly the wrong place to discuss future design :p why I didn't like it, is because that allows you to have an arbitrary led setup, which sounds perfect, except you are restricted to the 4 (for rtl93xx), 3 afaik for rtl839x? So now you have some very deep complexity in figuring this out. Ideally, you'd refer a led here via a phandle and which 'mode', but the mode bit doesn't exist yet (which you struggled with too). But again, wrong time/place. And something we should deff. have a brainstorming session over.